extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/src/main/java/org/apache/guacamole/auth/sqlserver/SQLServerEnvironment.java

```diff
@@ -0,0 +1,357 @@
+     * supported, these cannot be constants, as the legacy properties dictate
+     * the values that should be used in the absence of the correct properties.
+     */
+    private int DEFAULT_MAX_CONNECTIONS = 0;
+
+    /**
+     * The default value for the default maximum number of connections to be
+     * allowed to any one connection group. Note that, as long as the legacy
+     * "disallow duplicate" and "disallow simultaneous" properties are still
+     * supported, these cannot be constants, as the legacy properties dictate
+     * the values that should be used in the absence of the correct properties.
+     */
+    private int DEFAULT_MAX_GROUP_CONNECTIONS = 0;
+
+    /**
+     * The value for the sqlserver-driver property that triggers the use of
+     * the open source JTDS driver.
+     */
+    public final static String SQLSERVER_DRIVER_JTDS = "jtds";
+
+    /**
+     * The value for the sqlserver-driver property that triggers the use of
+     * the DataDirect JDBC driver.
+     */
+    public final static String SQLSERVER_DRIVER_DATADIRECT = "datadirect";
+
+    /**
+     * The value for the sqlserver-driver property that triggers the use of
+     * the older Microsoft JDBC driver.
+     */
+    public final static String SQLSERVER_DRIVER_MS = "microsoft";
+
+    /**
+     * The value for the sqlserver-driver property that triggers the use of
+     * the Microsoft JDBC driver.  This is the default.
+     */
+    public final static String SQLSERVER_DRIVER_MS_2005 = "microsoft2005";
+
+    /**
+     * Constructs a new SQLServerEnvironment, providing access to SQLServer-specific
+     * configuration options.
+     * 
+     * @throws GuacamoleException 
+     *     If an error occurs while setting up the underlying JDBCEnvironment
+     *     or while parsing legacy SQLServer configuration options.
+     */
+    public SQLServerEnvironment() throws GuacamoleException {
+
+        // Init underlying JDBC environment
+        super();
+
+        // Read legacy concurrency-related property
+        Boolean disallowSimultaneous = getProperty(SQLServerGuacamoleProperties.SQLSERVER_DISALLOW_SIMULTANEOUS_CONNECTIONS);
+        Boolean disallowDuplicate    = getProperty(SQLServerGuacamoleProperties.SQLSERVER_DISALLOW_DUPLICATE_CONNECTIONS);
+
+        // Legacy "simultaneous" property dictates only the maximum number of
+        // connections per connection
+        if (disallowSimultaneous != null) {
+
+            // Translate legacy property
+            if (disallowSimultaneous) {
+                DEFAULT_MAX_CONNECTIONS       = 1;
+                DEFAULT_MAX_GROUP_CONNECTIONS = 0;
+            }
+            else {
+                DEFAULT_MAX_CONNECTIONS       = 0;
+                DEFAULT_MAX_GROUP_CONNECTIONS = 0;
+            }
+
+            // Warn of deprecation
+            logger.warn("The \"{}\" property is deprecated. Use \"{}\" and \"{}\" instead.",
```

Comment:
The "disallow simultaneous connections", etc. properties continue to exist in the MySQL and PostgreSQL auth (with deprecation warnings) because they were at one time supported, and users are expected to migrate to the new properties when possible. This doesn't make sense here, though, since all SQL Server properties are new. If the properties shouldn't be used, they shouldn't be added.

Reply:
Okay, will remove them.

Reply:
Believe I got all this cleaned out.